hathor/transaction/storage/storage_protocol.py

```diff
@@ -0,0 +1,48 @@
+#  Copyright 2024 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from abc import abstractmethod
+from typing import Protocol
+
+from hathor.transaction import BaseTransaction, Block
+from hathor.types import VertexId
+
+
+class StorageProtocol(Protocol):
```

Comment:
`VertexStorageProtocol` or `VertexReducedStorageProtocol`

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/922/commits/edcaf5693c5460ba64649640636c49d5afbf0aa4

Reply:
Thanks!